pallets/asset-index/src/lib.rs

```diff
@@ -525,13 +528,18 @@ pub mod pallet {
         /// Creates IndexAssetData if entry with given assetID does not exist.
         /// Otherwise adds the units to the existing holding
         fn add_asset(
+            metadata: AssetMetadata,
             asset_id: &T::AssetId,
```

Comment:
This means `metadata` is required for every new `add_asset` call?
Can we extract this into a separate functions `set_metadata`?